lib/src/crypto_subtle.dart

```diff
@@ -73,22 +42,142 @@ Uint8List bigIntToUint8ListBigInteger(BigInt integer) {
   throw UnimplementedError('Only supports 65537 and 3 for now');
 }
 
-/// Wrapper for the [CryptoKey][1] type.
+/// The `window` object.
+@JS()
+external JSWindow get window;
+
+/// https://developer.mozilla.org/en-US/docs/Web/API/Window
+extension type JSWindow(JSObject _) implements JSObject {
+  /// https://developer.mozilla.org/en-US/docs/Web/API/crypto_property
+  external JSCrypto get crypto;
+}
+
+/// The `window.crypto` object.
 ///
-/// [1]: https://www.w3.org/TR/WebCryptoAPI/#cryptokey-interface
-class CryptoKey {
-  final Object _jsObj;
+/// https://www.w3.org/TR/WebCryptoAPI/#crypto-interface
+/// https://developer.mozilla.org/en-US/docs/Web/API/Crypto
+extension type JSCrypto(JSObject _) implements JSObject {
+  /// https://developer.mozilla.org/en-US/docs/Web/API/Crypto/subtle
+  external JSSubtleCrypto get subtle;
+
+  /// https://developer.mozilla.org/en-US/docs/Web/API/Crypto/getRandomValues
+  external JSTypedArray getRandomValues(JSTypedArray array);
+}
+
+/// The `window.crypto.subtle` object.
+///
+/// https://www.w3.org/TR/WebCryptoAPI/#subtlecrypto-interface
+/// https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto
+extension type JSSubtleCrypto(JSObject _) implements JSObject {
+  /// https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/encrypt
+  external JSPromise<JSArrayBuffer> encrypt(
+    JSAny algorithm,
+    JSCryptoKey key,
+    JSTypedArray data,
+  );
+
+  /// https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/decrypt
+  external JSPromise<JSArrayBuffer> decrypt(
+    JSAny algorithm,
+    JSAny key,
+    JSTypedArray data,
+  );
 
-  CryptoKey(this._jsObj);
+  /// https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/sign
+  external JSPromise<JSArrayBuffer> sign(
+    JSAny algorithm,
+    JSCryptoKey key,
+    JSTypedArray data,
+  );
+
+  /// https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/verify
+  external JSPromise<JSBoolean> verify(
+    JSAny algorithm,
+    JSCryptoKey key,
+    JSTypedArray signature,
+    JSTypedArray data,
+  );
 
+  /// https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest
+  external JSPromise<JSArrayBuffer> digest(
+    String algorithm,
+    JSTypedArray data,
+  );
+
+  /// https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/deriveBits
+  external JSPromise<JSArrayBuffer> deriveBits(
+    JSAny algorithm,
+    JSCryptoKey baseKey,
+    int length,
+  );
+
+  /// https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/generateKey
+  @JS('generateKey')
+  external JSPromise<JSCryptoKey> generateCreptoKey(
+    JSAny algorithm,
+    bool extractable,
+    JSArray<JSString> keyUsages,
+  );
+
+  /// https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/generateKey
+  @JS('generateKey')
+  external JSPromise<JSCryptoKeyPair> generateCreptoKeyPair(
```

Comment:
typo?

Reply:
Sorry, typo, I'll fix the two places together.

Reply:
It's typo. I fixed. [e9c8900](https://github.com/google/webcrypto.dart/pull/86/commits/e9c890031c926e235bed935a9bf590a3bde4e7e5)